source/ed/osm2ed.cpp

```diff
@@ -76,6 +76,15 @@ void ReadRelationsVisitor::relation_callback(uint64_t osm_id,
         if (it_level == accepted_levels.end()) {
             return;
         }
+        const auto& it_end_date = tags.find("end_date");
+        if (it_end_date != tags.end()) {
+            const boost::gregorian::date end_date(boost::gregorian::from_string(it_end_date->second));
```

Comment:
Not sure about the behavior if date is ill-formated.
I guess in that case we should keep the OSM object.

Reply:
it probably throws an exception, so ...

Reply:
If the date is ill-formated, boost::gregorian::date will set the value "not_a_date_time" for the instance and it can be checked with myDate.is_not_a_date().

But I check the behaviour with the operator '<' and '>': it's kind of weird because :
* not_a_date < valid_date == false
* not_a_date > valid_date == true

(source: https://www.boost.org/doc/libs/1_39_0/doc/html/date_time/gregorian.html)

I fix my patch to handle that case. Thanks for the heads-up :+1: